pkg/controller/resourcegraphdefinition/controller.go

```diff
@@ -85,10 +86,15 @@ func (r *ResourceGraphDefinitionReconciler) SetupWithManager(mgr ctrl.Manager) e
 		Complete(reconcile.AsReconciler[*v1alpha1.ResourceGraphDefinition](mgr.GetClient(), r))
 }
 
-func (r *ResourceGraphDefinitionReconciler) Reconcile(ctx context.Context, resourcegraphdefinition *v1alpha1.ResourceGraphDefinition) (ctrl.Result, error) {
-	rlog := r.log.WithValues("resourcegraphdefinition", types.NamespacedName{Namespace: resourcegraphdefinition.Namespace, Name: resourcegraphdefinition.Name})
+func (r *ResourceGraphDefinitionReconciler) Reconcile(ctx context.Context, req ctrl.Request) (ctrl.Result, error) {
+	rlog := r.log.WithValues("resourcegraphdefinition", req.NamespacedName)
 	ctx = log.IntoContext(ctx, rlog)
 
+	resourcegraphdefinition := &v1alpha1.ResourceGraphDefinition{}
+	if err := r.Get(ctx, req.NamespacedName, resourcegraphdefinition); err != nil {
+		return ctrl.Result{}, client.IgnoreNotFound(err)
+	}
+
```

Comment:
This is already happening in controller-runtime https://github.com/kubernetes-sigs/controller-runtime/blob/76067279145e76148cf27c01ea4ec63fed8e463b/pkg/reconcile/reconcile.go#L157-L165